smart_open/smart_open_lib.py

```diff
@@ -215,7 +219,7 @@ def smart_open(uri, mode="rb", **kw):
         raise TypeError('don\'t know how to handle uri %s' % repr(uri))
 
 
-def s3_open_uri(parsed_uri, mode, **kwargs):
+def s3_open_uri(parsed_uri, mode, ignore_extension=False, **kwargs):
```

Comment:
Should `ignore_extension` be an explicit parameter (or this should be part of `kwargs`), same question for `s3_open_key`, @mpenkov thought?